plugins/zfs/zfs_pool_dataset_count

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#%# family=auto
+
+: << EOF
+=head1 NAME
+zfs_pool_dataset_count - Outputs the count of zfs pools, datasets and snapshots.
+
+
+=head1 AUTHOR
+=over 4
+=item * Michael Grote
+=back
+
+=head1 LICENSE
+Unknown license
+
+=head1 MAGIC MARKERS
+=begin comment
+These magic markers are used by munin-node-configure when installing
+munin-node.
+=end comment
+  #%# family=auto
+  #%# capabilities=autoconf suggest
+=cut
+
+EOF
+
+if [ "$1" = "autoconf" ]; then
+    if ! command -v zpool &> /dev/null; then
+        echo "[ERROR] zpool could not be found!"
+    else
+        echo "yes"
+    fi
+    exit 0
+fi
+
+
+# lese alle pools, sed löscht die erste zeile
+list_pools=$(zpool list | sed 1d)
+
+
+if [ "$1" = "config" ]; then
+  # https://superuser.com/questions/284187/bash-iterating-over-lines-in-a-variable
+  while read -r line; do
+    # setze label <pool>.label <pool> snapshots
+    echo "$line" | awk '{print $1"_snapshot.label " $1 " snapshots"}'
+    echo "$line" | awk '{print $1"_dataset.label " $1 " datasets"}'
+  done <<< "$list_pools"
+  echo 'pools.label pools'
+  # setze optionen
+  echo 'graph_title zfs -  pool, dataset and snapshot count' # Titelzeile
+  echo 'graph_vlabel count' # Text links, hochkant
+  echo 'graph_category fs' # Kategorie
+  echo 'graph_args -l 0' # wertebegrenzer 0-100
+  exit 0
+fi
+
+# lese jede Zeile der Variable $list
+# tue für jede Zeile
+# "echo" die Zeile, wende awk drauf, Spalte $1/$7
+while read -r line; do
+  echo pools.value "$(zpool list | sed 1d | wc -l)"
+  # setze poolnamen
+  poolname=$(echo "$line" | awk '{ print $1 }')
+  # zähle snapshots
+  echo "$poolname""_snapshot.value" "$(zfs list -r -t snapshot "$poolname" | sed 1d | wc -l)"
```

Comment:
Here you may want to place curly braces around `poolname`, in order to avoid the confusing duplicated quotes.

Reply:
do you mean like `"${zfs list -r -t snapshot "$poolname" | sed 1d | wc -l}"`

Reply:
not quite :)

Here is yours:
```shell
echo "$poolname""_snapshot.value" "$(zfs list -r -t snapshot "$poolname" | sed 1d | wc -l)" 
```

This is my suggestion:
```shell
echo "${poolname}_snapshot.value" "$(zfs list -r -t snapshot "$poolname" | sed 1d | wc -l)" 
```

Reply:
got it, it is changed